docs/internals/contributing/writing-code/submitting-patches.txt

```diff
@@ -140,20 +140,62 @@ Regardless of the way you submit your work, follow these steps.
 .. _ticket tracker: https://code.djangoproject.com/
 .. _Development dashboard: https://dashboard.djangoproject.com/
 
-Non-trivial contributions
-=========================
+Contributions which require community feedback
+==============================================
 
-A "non-trivial" contribution is one that is more than a small bug fix. It's a
-change that introduces new Django functionality and makes some sort of design
-decision.
+A patch that introduces Django functionality and makes some sort of design
+decision requires a wider community discussion. Especially if the approach
+involves a deprecation or potentially introducing breaking changes.
+Please include evidence that the approach and alternatives have been discussed
+on the `Django Forum`_ or |django-developers| list.
 
-If you provide a non-trivial change, include evidence that alternatives have
-been discussed on the `Django Forum`_ or |django-developers| list.
+Here are a few approaches for gaining feedback from the community.
 
-If you're not sure whether your contribution should be considered non-trivial,
-ask on the ticket for opinions.
+Django Forum or the django-developers mailing list
+--------------------------------------------------
+
+You can make a proposal on the `Django Forum`_ or |django-developers| mailing list.
+
+You should explain the need for the change, go into details of the approach
+and discuss alternatives. Please include a link to such discussions in your patch.
+
+Third party package
+-------------------
+
+Django does not accept experimental features. If you need user feedback
+on a public interface, landing this into Django core is not the way to do it.
+Iterations must follow the
+:ref:`deprecation process<internal-release-deprecation-policy>`
+and can take months or even years for the users to see the benefit of such changes.
+
+Therefore, in such cases, it is better to create a third-party package which
+seeks user feedback. The public API can be iterated on and you can validate
+that there is a need for such a feature.
+
+Once this package becomes stable and there are clear benefits of
+incorporating aspects into Django core, starting a discussion on the
+`Django Forum`_ or |django-developers| mailing list would be the next step.
+
+Django Enhancement Proposal (DEP)
+---------------------------------
+
+Similar to Python’s PEPs, Django has `Django Enhancement Proposals`_ or DEPs.
+A DEP is a design document providing information to the Django community,
```

Comment:
Looks like the spell checker failed and we might need to add "DEP" to `docs/spelling_wordlist` (see https://docs.djangoproject.com/en/5.1/internals/contributing/writing-documentation/#spelling-check)

Reply:
Ah, Thank you. 
Will add it